impl/src/main/java/org/glassfish/soteria/identitystores/LdapIdentityStore.java

```diff
@@ -218,11 +228,24 @@ private CredentialValidationResult checkDirectBinding(UsernamePasswordCredential
         );
 
         // If this doesn't throw an exception internally, the caller dn exists and the password is correct
-        LdapContext ldapContext = createLdapContext(
+        LdapContext ldapContext = null;
+        try {
+          ldapContext = createLdapContext(
                 ldapIdentityStoreDefinition.url(),
                 callerDn,
                 new String(usernamePasswordCredential.getPassword().getValue())
-        );
+          );
+        } catch (Exception e) {
+          if (e instanceof AuthenticationException) {
```

Comment:
In this case we know the authentication problem is the user, so _INVALID_RESULT_ is correct, but we probably don't want to log this. There's only one _LOGGER_ call in the code before your changes, which is _WARNING_ about invalid configuration. Probably we should only log errors about bad configuration.